modules/preview-react/pill/stories/Pill.stories.mdx

```diff
@@ -0,0 +1,237 @@
+> with an ellipsis and render an OverflowTooltip on hover and focus. This max width can be changed
+> by providing a `maxWidth` prop on the Pill.
+
+<ExampleCodeBlock code={Basic} />
+
+### Interactive Pills
+
+All leading elements (icons or avatars) are intended to be descriptive, helping support the label.
+Do not rely on the leading element to indicate the interaction behavior. By default a Pill is
+considered interactive, therefore it's default variant is `default`.
+
+```tsx
+<Pill onClick={() => console.warn('clicked')}>
+  <Pill.Avatar />
+  Regina Skeltor
+</Pill>
+```
+
+#### Avatar
+
+The Avatar should appear before the text of a Pill. You have access to all Avatar props.
+
+<ExampleCodeBlock code={ClickablePillWithAvatar} />
+
+#### Icon
+
+The `Pill.Icon` icon will default to rendering a `plusIcon`. You can change this by providing a
+`icon` prop to `Pill.Icon`
+
+<ExampleCodeBlock code={ClickablePillWithIcon} />
+
+#### Count
+
+The count appears after the label. It is usually associate to the label. If you have a category, the
+count will dirrectly correlate to that category.
+
+<ExampleCodeBlock code={ClickablePillWithCount} />
+
+### Removable Pills
+
+Removable Pills display an `X` icon after the label. They have a smaller, more specific focus state
+and click target to be more intentional about their actions and to avoid unintended removal of a
+Pill.
+
+You can define a removable Pill by providing a `variant='removable'` prop.
+
+```tsx
+<Pill variant="removable" onDelete={() => console.warn('clicked')}>
+  Pink Shirts
+  <Pill.IconButton />
+</Pill>
+```
+
+In this case, we use a `Pill.IconButton` because the `X` becomes the focusable and clickable
+element.
+
+The default icon for `Pill.IconButton` is `xSmallIcon` but this can also be overwritten by passing
+an `icon` prop to `Pill.IconButton`
+
+<ExampleCodeBlock code={RemovablePill} />
+
+### List of Pills
+
+Pills can often represent multiple pieces of information such as a filtered list of categories or
+skills.
+
+In order to achieve this, use our `HStack` component to wrap each `Pill` and space them out
+accordingly.
+
+<ExampleCodeBlock code={ListOfPills} />
+
+## Components
+
+### Pill
+
+This component is the container component for a Pill. It also provides a React Context model for its
+sub components. Based on the `variant` prop this component will render different styled Pills.
+
+#### Usage
+
+Example of read only:
+
+```tsx
+<Pill variant="readOnly">This is a read only</Pill>
+```
+
+If you give this component a `variant='default'` prop it will render as an interactive Pill which
+accepts subcomponents.
+
+```tsx
+<Pill variant="default" onClick={() => console.log('clicked')}>
+  <Pill.Avatar /> Regina Skeltor
+</Pill>
+```
+
+If you give this component a `variant='removable'` it will render as a span and you can have a
+removable Pill that has a smaller click target to prevent users from accidentally delete something.
```

Comment:
```suggestion
If you set the `Pill` `variant` to `removable`, it will render a `<span>` element. You can then provide a `Pill.IconButton` that acts as the focus target. This creates a smaller, more intentional click target that prevents users from accidentally deleting an item.
```